src/rf/js/src/core/models.js

```diff
@@ -88,6 +108,53 @@ var Layer = Backbone.Model.extend({
         return this.isCompleted() || this.isFailed();
     },
 
+    getStatusByName: function(status) {
+        var start = 'status_' + status + '_start',
+            end = 'status_' + status + '_end',
+            error = 'status_' + status + '_error';
+
+        if (!this.layerStatusIsKnown(status)) {
```

Comment:
Yeah. This shouldn't ever happen but I don't really like seeing:
`this.layerStatusIsKnown(status);`
stand on its own and just crash out if the layer is not know. But I do like throwing the exception. I guess I could just have it do a console.error if the status is not found and allow it to return false. What do you think?